syncd/syncd_request_shutdown.cpp

```diff
@@ -16,7 +16,9 @@ int main(int argc, char **argv)
     static struct option long_options[] =
     {
         { "cold", no_argument, 0, 'c' },
-        { "warm", no_argument, 0, 'w' }
+        { "warm", no_argument, 0, 'w' },
+        { "fast", no_argument, 0, 'f' },
+        { "pre",  no_argument, 0, 'p' }, // Requesting pre shutdown
```

Comment:
>pre [](start = 11, length = 3)

Can we implement bash function wait_for_pre_shutdown_complete_or_fail inside syncd_request_shutdown? so when the process return, it's done.

Reply:
That is certainly possible. Take it in another review?